lms/services/lti_outcomes.py

```diff
@@ -61,6 +61,8 @@ def record_result(  # pylint:disable=no-self-use
             Defined as required by the LTI spec but is optional in Canvas if
             an `lti_launch_url` is set.
         :param pre_record_hook: Hook to allow modification of the request
+
+        :raise TypeError: if the given pre_record_hook returns a non-dict
```

Comment:
PyLint points out when a method raises an exception and you haven't documented that with a `:raise`. I doubt that it spots all exceptions that you might raise (even ones you fail to catch in methods you call directly or indirectly) but it finds some and that's useful